packages/IntegrationTest/windows/integrationtest/TestHostHarness.cpp

```diff
@@ -165,7 +165,14 @@ winrt::fire_and_forget TestHostHarness::TimeoutOnInactivty(winrt::weak_ref<TestT
   if (auto strongTransaction = transaction.get()) {
     if (auto strongThis = weakThis.get()) {
       if (m_currentTransaction == strongTransaction) {
-        HandleHostAction(m_currentTransaction->OnTimeout());
+
+        // The timer runs on a threadpool, so it could jump threads, when it does repost
+        if (m_dispatcher.HasThreadAccess()) {
+          HandleHostAction(m_currentTransaction->OnTimeout());
+        } else {
+          m_dispatcher.Post(
+              [strongThis]() noexcept { strongThis->HandleHostAction(strongThis->m_currentTransaction->OnTimeout()); });
```

Comment:
We probably want to dispatch before we compare transactions. I have a slightly different solution here https://github.com/microsoft/react-native-windows/pull/6109

Reply:
As discussed offline, I'll wait for yours to merge...

Reply:
Would you mind signing off on the other one? 😛 .

Reply:
Merged